src/test/java/wooteco/subway/domain/SectionsTest.java

```diff
@@ -0,0 +1,166 @@
+        Section splitSection = new Section(6L, 2L, 6);
+        sections.add(section);
+
+        assertAll(
+            () -> assertThat(sections.hasSection(section)).isTrue(),
+            () -> assertThat(sections.hasSection(splitSection)).isTrue()
+        );
+    }
+
+    @DisplayName("상행이 동일한 구간이 존재한지만 새로 입력하는 구간의 길이가 크거나 같다면 예외가 발생한다.")
+    @Test
+    void addSectionInsideSections_up_fail() {
+        Section section = new Section(1L, 6L, 10);
+
+        assertThatThrownBy(() -> sections.add(section))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("[ERROR] 역 사이에 새로운 역을 등록할 경우 기존 역 사이 길이보다 크거나 같으면 등록할 수 없습니다.");
+    }
+
+    @DisplayName("동일한 상행 지하철역이 존재한다면 기존 구간을 분리하여 저장한다.")
+    @Test
+    void addSectionInsideSections_down() {
+        Section section = new Section(6L, 2L, 4);
+        Section splitSection = new Section(1L, 6L, 6);
+        sections.add(section);
+
+        assertAll(
+            () -> assertThat(sections.hasSection(section)).isTrue(),
+            () -> assertThat(sections.hasSection(splitSection)).isTrue()
+        );
+    }
+
+    @DisplayName("하행이 동일한 구간이 존재한지만 새로 입력하는 구간의 길이가 크거나 같다면 예외가 발생한다.")
+    @Test
+    void addSectionInsideSections_down_fail() {
+        Section section = new Section(6L, 2L, 10);
+
+        assertThatThrownBy(() -> sections.add(section))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("[ERROR] 역 사이에 새로운 역을 등록할 경우 기존 역 사이 길이보다 크거나 같으면 등록할 수 없습니다.");
+
+    }
+
+    @DisplayName("상행 종점을 제거할 경우 다음에 오는 역이 종점이 된다.")
+    @Test
+    void removeUpStation() {
+        Section section = new Section(1L, 2L, 10);
+        long stationId = 1L;
+
+        sections.remove(stationId);
+        assertAll(
+            () -> assertThat(sections.getSections().size()).isEqualTo(2),
+            () -> assertThat(sections.hasSection(section)).isFalse()
+        );
+    }
+
+    @DisplayName("하행 종점을 제거할 경우 이전에 있는 역이 종점이 된다.")
+    @Test
+    void removeDownStation() {
+        Section section = new Section(3L, 4L, 10);
+        long stationId = 4L;
+
+        sections.remove(stationId);
+        assertAll(
+            () -> assertThat(sections.getSections().size()).isEqualTo(2),
+            () -> assertThat(sections.hasSection(section)).isFalse()
+        );
+    }
+
+    @DisplayName("구간 중간에 있는 역 제거할 경우 각 양 옆의 구간의 합으로 재배치된다.")
+    @Test
+    void removeWayPointStation() {
+        Section section1 = new Section(2L, 3L, 10);
+        Section section2 = new Section(1L, 2L, 10);
+        Section newSection = new Section(1L, 3L, 20);
+        long stationId = 2L;
+
+        sections.remove(stationId);
+        assertAll(
+            () -> assertThat(sections.getSections().size()).isEqualTo(2),
+            () -> assertThat(sections.hasSection(section1)).isFalse(),
+            () -> assertThat(sections.hasSection(section2)).isFalse(),
+            () -> assertThat(sections.hasSection(newSection)).isTrue()
+        );
+    }
```

Comment:
```suggestion
    @DisplayName("구간 중간에 있는 역 제거할 경우 각 양 옆의 구간의 합으로 재배치된다.")
    @Test
     void WhenRemoveMiddleStation_ThenMergeSideStations() {
        Long middelStationId = 2L;
        Section formerSection = new Section(1L, middleStationId, 10);
        Section laterSection = new Section(middleStationId, 3L, 10);
        Section mergedSection = new Section(1L, 3L, 20);
        
        sections.remove(stationId);
        assertAll(
            () -> assertThat(sections.getSections().size()).isEqualTo(2),
            () -> assertThat(sections.hasSection(formerSection)).isFalse(),
            () -> assertThat(sections.hasSection(laterSection)).isFalse(),
            () -> assertThat(sections.hasSection(mergedSection)).isTrue()
        );
    }
```

지극히 주관적인 내용이라 참고만 부탁할게요 바니. 코드를 보는 사람이 이해할 수 있도록 변수 명에 신경써봤어요.

Reply:
제가 느끼기에도 훨씬 이해하기 쉬운 코드라고 생각이 들었습니다.
테스트 코드 또한 이해할 수 있도록 변수명에 신경쓰도록 하겠습니다!! ㅎㅎ